internal/db/fetcher/document.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package fetcher
+
+import (
+	"bytes"
+	"context"
+
+	dsq "github.com/ipfs/go-datastore/query"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore/iterable"
+	"github.com/sourcenetwork/defradb/internal/db/base"
+	"github.com/sourcenetwork/defradb/internal/keys"
+)
+
+// document is the type responsible for fetching documents from the datastore.
+//
+// It does not filter the data in any way.
+type document struct {
```

Comment:
suggestion: Similar renaming suggestion as above.

Reply:
same comment as above, I don't think `document` is the best naming here, too generic/general and could be interpreted in a myriad of ways (which I initially did). 

In addition to my above comment, from a purely gramatical POV, the `fetcher` is a verb indicating *action* of some kind, which makes sense since the fetchers are a system of pulling and processing key values into a consistent output form. Where as `document` is a noun and indicates a static object. (This grammar argument isn't my primary concern with the naming, just an observation)